graph mode: add hardswish inplace handling

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40284 graph mode: add hardswish inplace handling**
* #40111 quantized elu: graph mode handling
* #40104 quantized elu: eager mode QAT handling
* #40103 quantized elu: eager mode static handling
* #40100 quantized elu: require observation

Summary:

Adds graph mode handling for inplace hardswish, and test coverage for functional hardswish.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_hardswish
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22140628](https://our.internmc.facebook.com/intern/diff/D22140628)

vkuzo committed:
graph more: add hardswish inplace handling

Summary:

WIP

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_hardswish
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: add hardswish inplace handling"


Summary:

WIP

need to fix inplace before landing.  graph comparison: https://www.internalfb.com/intern/paste/P133715227/

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_hardswish
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/testing/_internal/common_quantization.py:
```diff
@@ -282,8 +283,12 @@ def checkGraphModeOp(self, module, data, quantized_op, tracing=False, debug=Fals
             else:
+                # module under test can contain in-place ops, and we depend on
+                # input data staying constant for comparisons
+                data_copy = copy.deepcopy(data)
```
makes sense, wonder why we didn't discover this in other tests

jerryzh168 replied:
oh, maybe this is the first one modifying input data

jerryzh168 commented on test/quantization/test_quantize_script.py:
```diff
@@ -2082,12 +2082,24 @@ def forward(self, x):
         for tracing in [True, False]:
-            m = self.checkGraphModeOp(hardswish, data, "quantized::hardswish", tracing)
-            FileCheck().check_not("aten::hardswish") \
-                       .run(m.graph)
+            for m in modules:
```
you can use itertools to reduce identation, see other test cases in the file

jerryzh168 reviewed (approved):
LG

vkuzo committed:
Update on "graph mode: add hardswish inplace handling"


Summary:

Adds graph mode handling for inplace hardswish, and test coverage for functional hardswish.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_hardswish
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22140628](https://our.internmc.facebook.com/intern/diff/D22140628)

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: add hardswish inplace handling"


Summary:

Adds graph mode handling for inplace hardswish, and test coverage for functional hardswish.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_hardswish
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22140628](https://our.internmc.facebook.com/intern/diff/D22140628)

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: add hardswish inplace handling"


Summary:

Adds graph mode handling for inplace hardswish, and test coverage for functional hardswish.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_hardswish
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22140628](https://our.internmc.facebook.com/intern/diff/D22140628)

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: add hardswish inplace handling"


Summary:

Adds graph mode handling for inplace hardswish, and test coverage for functional hardswish.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_hardswish
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22140628](https://our.internmc.facebook.com/intern/diff/D22140628)

[ghstack-poisoned]
